components/post_view/channel_intro_message/messages/standard.tsx

```diff
@@ -0,0 +1,186 @@
+
+    const memberMessage = useMemo(() => {
+        if (isArchivedChannel(channel)) {
+            return null;
+        }
+        if (channel.type === Constants.PRIVATE_CHANNEL) {
+            return (
+                <FormattedMessage
+                    id='intro_messages.onlyInvited'
+                    defaultMessage=' Only invited members can see this private channel.'
+                />
+            );
+        }
+        return (
+            <FormattedMessage
+                id='intro_messages.anyMember'
+                defaultMessage=' Any member can join and read this channel.'
+            />
+        );
+    }, [channel]);
+
+    const createMessage = useMemo(() => {
+        const date = (
+            <FormattedDate
+                value={channel.create_at}
+                month={getMonthLong(locale)}
+                day='2-digit'
+                year='numeric'
+            />
+        );
+        if (!creatorName) {
+            if (channel.type === Constants.PRIVATE_CHANNEL) {
+                return (
+                    <FormattedMessage
+                        id='intro_messages.noCreatorPrivate'
+                        defaultMessage='This is the start of the {name} private channel, created on {date}.'
+                        values={{name: uiName, date}}
+                    />
+                );
+            }
+            if (channel.type === Constants.OPEN_CHANNEL) {
+                return (
+                    <FormattedMessage
+                        id='intro_messages.noCreator'
+                        defaultMessage='This is the start of the {name} channel, created on {date}.'
+                        values={{name: uiName, date}}
+                    />
+                );
+            }
+        }
+
+        if (channel.type === Constants.PRIVATE_CHANNEL) {
+            return (
+                <span>
+                    <FormattedMessage
+                        id='intro_messages.creatorPrivate'
+                        defaultMessage='This is the start of the {name} private channel, created by {creator} on {date}.'
+                        values={{
+                            name: uiName,
+                            creator: creatorName,
+                            date,
+                        }}
+                    />
+                </span>
+            );
+        }
+
+        if (channel.type === Constants.OPEN_CHANNEL) {
+            return (
+                <span>
+                    <FormattedMessage
+                        id='intro_messages.creator'
+                        defaultMessage='This is the start of the {name} channel, created by {creator} on {date}.'
+                        values={{
+                            name: uiName,
+                            creator: creatorName,
+                            date,
+                        }}
+                    />
+                </span>
+            );
+        }
+        return null;
```

Comment:
@michelengelen, What do you think of rewriting these if blocks like the following suggestion instead of returning a `FormattedMessage` inside each if block?

```suggestion
        let id;
        let defaultMessage;
        if (channel.type === Constants.PRIVATE_CHANNEL) {
            id = t('intro_messages.creatorPrivate');
            defaultMessage = t('This is the start of the {name} private channel, created by {creator} on {date}.');
        } else if (channel.type === Constants.OPEN_CHANNEL) {
            id = t('intro_messages.creator');
            defaultMessage = t('This is the start of the {name} channel, created by {creator} on {date}.');
        }
        return id && defaultMessage && (
            <span>
                <FormattedMessage
                    id={t(id}
                    defaultMessage={defaultMessage}
                    values={{
                        name: uiName,
                        creator: creatorName,
                        date,
                    }}
                />
            </span>
        );

```